code/datums/storage/storage.dm

```diff
@@ -527,6 +527,8 @@ GLOBAL_LIST_EMPTY(cached_storage_typecaches)
 /datum/storage/proc/attempt_remove(obj/item/thing, atom/remove_to_loc, silent = FALSE)
 	SHOULD_NOT_SLEEP(TRUE)
 
+	SEND_SIGNAL(parent, COMSIG_STORAGE_REMOVED_ITEM, thing, remove_to_loc, silent)
```

Comment:
Interestingly you are changing the order of operations on this,

Before it updated / qdeleted AFTER removing the items from contents, now, it's before. 

Kinda worried this might be causing bugs

Reply:
i can either convert it to a post removal signal or just ADD a post removal signal, i perfer the second since it allows for more future functionality but the 1st can be done too

Reply:
anjd youre right, it did cause it to not properly qdel itself, so ill change it